code/modules/mob/living/carbon/human/human_update_icons.dm

```diff
@@ -600,8 +600,17 @@ There are several things that need to be remembered:
 		drop_all_held_items()
 		return
 
+	if(held_items[1] == null && held_items[2] == null && hud_used)
+		for(var/atom/movable/screen/craft/crafting_button in hud_used.static_inventory)
+			crafting_button.icon = hud_used.ui_style
+			crafting_button.icon_state = initial(crafting_button.icon_state)
+
 	var/list/hands = list()
 	for(var/obj/item/worn_item in held_items)
+		if(((worn_item.type in GLOB.crafting_recipe_items) || ((worn_item.parent_type in GLOB.crafting_recipe_items) && !(worn_item.type in GLOB.crafting_recipe_items_blacklist))) && hud_used)
```

Comment:
These should be typecaches.
the `in` keyword requires iterating the whole list, whereas typecaches use associated lists, which are logarithmic time, can' remember which